EasyReflectometry/sample/assemblies/gradient_layer.py

```diff
@@ -63,34 +67,28 @@ def thickness(self) -> float:
         """
         :return: Thickness of the gradient layer
         """
-        # Layer 0 is the deciding layer as set in _apply_thickness_constraints
-        return self.layers[0].thickness.raw_value * self._discretisation_elements
+        return self.top_layer.thickness.raw_value * self._discretisation_elements
 
     @thickness.setter
     def thickness(self, thickness: float) -> None:
         """
         :param thickness: Thickness of the gradient layer
         """
-        # Layer 0 is the deciding layer as set in _apply_thickness_constraints
-        self.layers[0].thickness.value = thickness / self._discretisation_elements
+        self.top_layer.thickness.value = thickness / self._discretisation_elements
```

Comment:
value of `discretisation_elements` can be set by the caller, meaning we can't trust it to be non-zero...

Reply:
I am not sure I understand.  In the constructor L38 it is ensured that `self._discretisation_elements` > 2

Reply:
Old people's blindness. Yes, there is a check